Level 2

x-borz opened the pull request:
Верстка

x-borz committed:
feat. Добавить .gitignore

x-borz committed:
feat. Добавить каркас react-приложения

x-borz committed:
feat. Добавить файловую структуру приложения

x-borz committed:
feat. Добавить .editorconfig

x-borz committed:
feat. Сверстал часть компонентов

x-borz committed:
fix footer

x-borz committed:
feat. Добавил новые компоненты

Commit:
fix

Commit:
feat. landing адаптив 768

Commit:
fix

Commit:
feat. landing адаптив 320px, еще не весь

x-borz committed:
feat. landing адаптив 320px

x-borz committed:
fix классы по БЭМ

Commit:
feat. Movies & SavedMovies pages

Commit:
feat. компонент filter-checkbox

Commit:
feat. компонент filter-checkbox

x-borz committed:
fix movies 1280px

x-borz committed:
feat movies & saved movies res 768px

x-borz committed:
feat movies & saved movies res 320px

Commit:
fix submit btn

Commit:
feat. Добавить компонент NotFound

Commit:
feat. Добавить компонент Register, Login

Commit:
feat. Добавить компонент Register, Login. Res 768, 320

Commit:
feat. Добавить компонент Profile 1280

Commit:
feat. Добавить компонент Profile 768

Commit:
feat. Добавить компонент Profile 320

Commit:
fix html title

Commit:
feat. Добавить роутинг

Commit:
feat. Добавить Header 1280px MAIN

Commit:
feat. header main 320, 768

Commit:
feat. header

Commit:
feat. header

Commit:
feat. header

Commit:
feat. Menu component

Commit:
fix

Commit:
feat. add component Preloader

Commit:
feat. add component Preloader

Commit:
feat. popup для редактирования данных профиля

Commit:
refactoring

Commit:
feat. Добавил плавность открытия/закрытия бокового меню, закрытие меню по нажатию Escape

Commit:
fix

Commit:
fix

Commit:
fix

Commit:
fix

Commit:
fix

coolswood commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- [x] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.
`На расширении 320рх цвет фонового блока techs не соответствует макету. Обратите внимание что цвет в макете меняется`

- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вклxaадке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
`portfolio__link-row это должна быть единая ссылка (при клике и на текст и на иконку)`

- [x] <!--5.55--> Именование классов сделаны по БЭМ.
- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [x] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфоxм, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
	- [x] <!--3.38--> разметка закчлючена в ( ), лишние пустые <div> для этого не используются;
	- [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
	- [x] <!--3.38--> директория images/ с картинками,
	- [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- [x] <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете. 
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
`Элементы формы поиска фильмов не выделяются, когда на них установлен ФОКУС.`
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
`У формы поиска фильмов нет свойства required`
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации
- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

### Рекомендации
`Неправильно задан lang`

## Количество баллов:  74

Commit:
fix.
1. lang='ru'
2. Форма поиска фильмов: добавлено свойства required
3. Форма поиска фильмов: добавлено выделение элементов при фокусе на них
4. На разрешении 320px для блока Techs меняется фон (в соотвествии с макетом)

Commit:
fix.
5. В портфолио сделал ссылку единой: тест + маркер

coolswood commented:
# Резюме по работе:
## Верстка и JSX

### Критерии, влияющие на работоспособность
- [x] <!--12--> Все блоки и элементы из макета сверстаны. Размеры и расположение не отличаются от макета на более, чем на 30px на десктопной версии и на 10px на остальных размерах экрана, для которых подготовлен макет. Отличия в размерах не должны визуально противоречить макету и выбиваться из заложенной в макет сетки, в этом случае требуется более строгое соответствие размерам макета.

- [x] <!--10.55--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--9.55--> Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вклxaадке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.

- [x] <!--5.55--> Именование классов сделаны по БЭМ.
- [x] <!--8.55--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
- [x] <!--3.38--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфоxм, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--3.38--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--3.38--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--3.38--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--3.38--> Работа модальных окон настроена.
- [x] Разметка портирована в JSX:
	- [x] <!--3.38--> разметка закчлючена в ( ), лишние пустые <div> для этого не используются;
	- [x] <!--3.38--> разметка вынесена в соответствующие ей компоненты.
- [x] В проекте есть:
	- [x] <!--3.38--> директория images/ с картинками,
	- [x] <!--3.38--> директория components/ с JS и CSS-файлами компонентов,
	- [x] <!--3.38--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете. 
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> Иконки экспортированы в формате SVG.
- [x] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
- [x] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
- [x] <!--2.14--> reset.css запрещен.

### Рекомендации
- [x] <!--1.68--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов:  100
